### PR TITLE
create a Default Dialog to use in Gse

### PR DESCRIPTION
**Pull request by nashousni**

### CODE REVIEW

File: gse-util/src/main/java/com/powsybl/gse/util/NodeChooser.java [hunk 1]
```diff
@@ -807,32 +807,23 @@ private void savePreferences() {
     public static <N, F extends N, D extends N, T extends N> Optional<T> showAndWaitDialog(
             TreeModel<N, F, D> treeModel, Window window, AppData appData, GseContext context,
             BiPredicate<N, TreeModel<N, F, D>> filter, Set<String>... openedProjectsList) {
-        Dialog<T> dialog = new Dialog<>();
+        Dialog<T> dialog = null;
         try {
-            dialog.setTitle(RESOURCE_BUNDLE.getString("OpenFile"));
-            dialog.getDialogPane().getButtonTypes().addAll(ButtonType.OK, ButtonType.CANCEL);
             NodeChooser<N, F, D, T> nodeChooser = new NodeChooser<>(window, treeModel, appData, context, filter, openedProjectsList);
+            dialog = new GseDialog<>(RESOURCE_BUNDLE.getString("OpenFile"), nodeChooser, window, nodeChooser.selectedNodeProperty().isNull(), buttonType -> buttonType == ButtonType.OK ? nodeChooser.selectedNodeProperty().get() : null);
```
Comment (mathbagu) [inline gse-util/src/main/java/com/powsybl/gse/util/NodeChooser.java]: Extract the lambda used as a Callback to a variable to improve readability.

File: gse-util/src/main/java/com/powsybl/gse/util/NewScriptButton.java [hunk 2]
```diff
@@ -85,14 +78,10 @@ public static Button create(ProjectNode node, ObjectProperty<ModificationScript>
...
         pane.add(nameTextField.getFileAlreadyExistsLabel(), 0, 2, 4, 1);
-        dialog.getDialogPane().setContent(pane);
         BooleanBinding okProperty = folderSelectionPane.nodeProperty().isNotNull()
                                                        .and(nameTextField.okProperty());
-        button.disableProperty().bind(okProperty.not());
-        dialog.setResizable(true);
-        dialog.setResultConverter(buttonType -> buttonType == ButtonType.OK ? Boolean.TRUE : Boolean.FALSE);
         Platform.runLater(nameTextField.getInputField()::requestFocus);
-        return dialog;
+        return new GseDialog<>(RESOURCE_BUNDLE.getString("DialogTitle"), pane, window, okProperty.not(), buttonType -> buttonType == ButtonType.OK ? Boolean.TRUE : Boolean.FALSE);
```
Comment (mathbagu) [inline gse-util/src/main/java/com/powsybl/gse/util/NewScriptButton.java]: Extract the lambda used as a Callback to a variable to improve readability.

File: gse-util/src/main/java/com/powsybl/gse/util/NewFolderPane.java [hunk 3]
```diff
@@ -96,20 +96,16 @@ private F createFolder() {
...
-            dialog.setTitle(RESOURCE_BUNDLE.getString("NewFolder"));
-            dialog.getDialogPane().getButtonTypes().addAll(ButtonType.OK, ButtonType.CANCEL);
             NewFolderPane<F> newProjectPane = new NewFolderPane<>(folderCreator, folderUnique);
             newProjectPane.setPrefSize(350, 100);
-            dialog.getDialogPane().lookupButton(ButtonType.OK).disableProperty().bind(newProjectPane.validatedProperty().not());
-            dialog.getDialogPane().setContent(newProjectPane);
-            dialog.setResizable(true);
-            dialog.initOwner(window);
-            dialog.setResultConverter(buttonType -> buttonType == ButtonType.OK ? newProjectPane.createFolder() : null);
+            dialog = new GseDialog<>(RESOURCE_BUNDLE.getString("NewFolder"), newProjectPane, window, newProjectPane.validatedProperty().not(), buttonType -> buttonType == ButtonType.OK ? newProjectPane.createFolder() : null);
```
Comment (mathbagu) [inline gse-util/src/main/java/com/powsybl/gse/util/NewFolderPane.java]: Extract the lambda used as a Callback to a variable to improve readability.

File: gse-util/src/main/java/com/powsybl/gse/util/GseDialog.java [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+import javafx.util.Callback;
+
+/**
+ * @author Nassirou Nambiema <nassirou.nambiena at rte-france.com>
+ */
+public class GseDialog<F> extends Dialog<F> {
+
+    public GseDialog(String title, Node content, Window window, ObservableValue<? extends Boolean> observable, Callback<ButtonType, F> value) {
+        setTitle(title);
+        getDialogPane().getButtonTypes().addAll(ButtonType.OK, ButtonType.CANCEL);
```
Comment (mathbagu) [inline gse-util/src/main/java/com/powsybl/gse/util/GseDialog.java]: You should probably add another constructor for people who want another buttons configuration.
This constructor should call the new one, with the default values.

Comment (nashousni) [reply gse-util/src/main/java/com/powsybl/gse/util/GseDialog.java]: code OK

File: gse-util/src/main/java/com/powsybl/gse/util/NewFolderPane.java (hunk 3 above)
Comment (nashousni) [reply gse-util/src/main/java/com/powsybl/gse/util/NewFolderPane.java]: updated

File: gse-util/src/main/java/com/powsybl/gse/util/NewScriptButton.java (hunk 2 above)
Comment (nashousni) [reply gse-util/src/main/java/com/powsybl/gse/util/NewScriptButton.java]: updated

File: gse-util/src/main/java/com/powsybl/gse/util/NodeChooser.java (hunk 1 above)
Comment (nashousni) [reply gse-util/src/main/java/com/powsybl/gse/util/NodeChooser.java]: updated

**Comment by mathbagu** [comment]
One have to setup the quality gate in powsybl-gse sonar profile to be able to merge PR without unit-test

**Pull request merged**